pulsar-client-cpp/lib/PartitionedProducerImpl.cc

```diff
@@ -161,7 +182,13 @@ void PartitionedProducerImpl::sendAsync(const Message& msg, SendCallback callbac
     }
     // find a producer for that partition, index should start from 0
     ProducerImplPtr producer = producers_[partition];
+
+    if (conf_.getLazyStartPartitionedProducers() && !producer->isStarted()) {
+        producer->start();
```

Comment:
is producer::start() idempotent?

Reply:
It isn't but this code is protected by the producersLock so it can't be called twice concurrently.